### PR TITLE
MGMT-15397: Add Nutanix post installation step to CI

### PR DESCRIPTION
**Pull request by eliorerz**
/cc @eliorerz

### CODE REVIEW

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

File: ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh [hunk 1]
```diff
@@ -51,10 +51,209 @@ stringData:
...
+  config: "{\n\t\"prismCentral\": {\n\t\t\"address\": \"${NUTANIX_ENDPOINT}\",\n\t\t\"port\":${NUTANIX_PORT},\n\t\t\"credentialRef\": {\n\t\t\t\"kind\": \"Secret\",\n\t\t\t\"name\": \"nutanix-credentials\",\n\t\t\t\"namespace\": \"openshift-cloud-controller-manager\"\n\t\t}\n\t},\n\t\"topologyDiscovery\": {\n\t\t\"type\": \"Prism\",\n\t\t\"topologyCategories\": null\n\t},\n\t\"enableCustomLabeling\": true\n}\n"
+EOF
+fi
+
+# Create Operator Group
+cat <<EOF | oc create -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  generateName: openshift-cluster-csi-drivers
```
Comment (wolfganghuse) [inline ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:76]: generateName will append a random five character appendix, so it is recommended to have a trailing dash in the Name-Suffix.

File: ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh [hunk 2]
```diff
@@ -51,10 +51,209 @@ stringData:
...
-until \
-  oc wait --all=true clusteroperator --for='condition=Available=True' >/dev/null && \
-  oc wait --all=true clusteroperator --for='condition=Progressing=False' >/dev/null && \
-  oc wait --all=true clusteroperator --for='condition=Degraded=False' >/dev/null;  do
-    echo "$(date --rfc-3339=seconds) Clusteroperators not yet ready"
-    sleep 1s
+version=$(oc version | grep -oE 'Server Version: ([0-9]+\.[0-9]+)' | sed 's/Server Version: //')
+resource_timeout_seconds=300
+
+# Do the following if OCP version is >=4.13
```
Comment (osherdp) [inline ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:57]: why we need this configmap only for 4.13?

File: ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh [hunk 3]
```diff
@@ -51,10 +51,209 @@ stringData:
...
+if [[ $(echo -e "4.13\n$version" | sort -V | tail -n 1) == "$version" ]]; then
+  echo "Found OCP version greater or equal to 4.13, creating cloud-provider-config ConfigMap"
+  cat <<EOF | oc apply -f -
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: cloud-provider-config
+  namespace: openshift-config
+data:
+  config: "{\n\t\"prismCentral\": {\n\t\t\"address\": \"${NUTANIX_ENDPOINT}\",\n\t\t\"port\":${NUTANIX_PORT},\n\t\t\"credentialRef\": {\n\t\t\t\"kind\": \"Secret\",\n\t\t\t\"name\": \"nutanix-credentials\",\n\t\t\t\"namespace\": \"openshift-cloud-controller-manager\"\n\t\t}\n\t},\n\t\"topologyDiscovery\": {\n\t\t\"type\": \"Prism\",\n\t\t\"topologyCategories\": null\n\t},\n\t\"enableCustomLabeling\": true\n}\n"
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:67]:
```suggestion
  config: |
    {
      "prismCentral": {
        "address": "${NUTANIX_ENDPOINT}",
        "port": ${NUTANIX_PORT},
        "credentialRef": {
          "kind": "Secret",
          "name": "nutanix-credentials",
          "namespace": "openshift-cloud-controller-manager"
        }
      },
      "topologyDiscovery": {
        "type": "Prism",
        "topologyCategories": null
      },
      "enableCustomLabeling": true
    }
```
can we make it a bit more readable?

Comment (eliorerz) [reply ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:67]: This is how it's saved in openshift and that didn't worked when I tried that. I'll try to insist more making it more readable

File: ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh (hunk 2 above)
Comment (eliorerz) [reply ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:57]: I'll add it to the comment

File: ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh (hunk 1 above)
Comment (eliorerz) [reply ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh:76]: Yeah, I saw that, thanks

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-4-13-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-nutanix-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-nutanix-assisted

**Review by adriengentil — APPROVED** [review]

**Pull request merged**